server/src/main/scala/za/co/absa/atum/server/api/service/PartitioningServiceImpl.scala

```diff
@@ -61,6 +58,46 @@ class PartitioningServiceImpl(partitioningRepository: PartitioningRepository)
       }
   }
 
+  override def getPartitioningCheckpoints(checkpointQueryDTO: CheckpointQueryDTO): IO[ServiceError, Seq[CheckpointDTO]] = {
+    repositoryCall(
+      partitioningRepository.getPartitioningCheckpoints(checkpointQueryDTO), "getPartitioningCheckpoint"
+    ).map {
+      checkpointMeasurementsSeq =>
+        checkpointMeasurementsSeq.map { cm =>
+          CheckpointDTO(
+            id = cm.idCheckpoint,
+            name = cm.checkpointName,
+            author = cm.author,
+            measuredByAtumAgent = cm.measuredByAtumAgent,
+            partitioning = checkpointQueryDTO.partitioning,
+            processStartTime = cm.checkpointStartTime,
+            processEndTime = cm.checkpointEndTime,
+            measurements = Set(
+              MeasurementDTO(
+                measure = MeasureDTO(
+                  measureName = cm.measureName,
+                  measuredColumns = cm.measureColumns
+                ),
+                result = MeasureResultDTO(
+                  mainValue = MeasureResultDTO.TypedValue(
+                    value = cm.measurementValue.hcursor.downField("value").as[String].getOrElse(""),
+                    valueType = cm.measurementValue.hcursor.downField("valueType").as[String].getOrElse("") match {
+                      case "String" => MeasureResultDTO.ResultValueType.String
+                      case "Long" => MeasureResultDTO.ResultValueType.Long
+                      case "BigDecimal" => MeasureResultDTO.ResultValueType.BigDecimal
+                      case "Double" => MeasureResultDTO.ResultValueType.Double
+                      case _ => MeasureResultDTO.ResultValueType.String
```

Comment:
this is not necessary, and I don't recommend it either - we have (even you added) custom SerDe of the `ResultValueType` and related types. Just import it here, it will work. Did you see https://github.com/AbsaOSS/atum-service/pull/195 ? If not, it's what I did basically since our tickets are very similar